src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -333,11 +350,54 @@ void Process ()
 						_context.MarkingHelpers.MarkExportedType (exported, assembly.MainModule, di);
 					}
 				}
+
+				MarkFullyPreservedAssemblies ();
```

Comment:
I wonder if it would be better to make this return `bool`, just like `ProcessPrimaryQueue` and similar and call it from the `while` condition. The advantage would be that it would be done only once everything else was processed. Currently it runs every time we finish one of the queues. So very likely more often.

We could probably do something similar for the type forwards stuff above as well. Both of these operations are not order sensitive, we just need to run them eventually and then process all queues ones they're done. But it would be more optimal to run them late so that we don't have to repeat them that often.

Reply:
Changed to return a bool